scripts/db.dropcolumn/db.dropcolumn.py

```diff
@@ -80,30 +80,42 @@ def main():
         return 0
 
     if driver == "sqlite":
-        # echo "Using special trick for SQLite"
-        # http://www.sqlite.org/faq.html#q13
-        colnames = []
-        coltypes = []
-        for f in gscript.db_describe(table)["cols"]:
-            if f[0] != column:
-                colnames.append(f[0])
-                coltypes.append("%s %s" % (f[0], f[1]))
-
-        colnames = ", ".join(colnames)
-        coltypes = ", ".join(coltypes)
-
-        cmds = [
-            "BEGIN TRANSACTION",
-            "CREATE TEMPORARY TABLE ${table}_backup(${coldef})",
-            "INSERT INTO ${table}_backup SELECT ${colnames} FROM ${table}",
-            "DROP TABLE ${table}",
-            "CREATE TABLE ${table}(${coldef})",
-            "INSERT INTO ${table} SELECT ${colnames} FROM ${table}_backup",
-            "DROP TABLE ${table}_backup",
-            "COMMIT",
-        ]
-        tmpl = string.Template(";\n".join(cmds))
-        sql = tmpl.substitute(table=table, coldef=coltypes, colnames=colnames)
+        sqlite3_version = gscript.read_command(
+            "db.select",
+            sql="SELECT sqlite_version();",
+            flags="c",
+            database=database,
+            driver=driver,
+        ).split(".")[0:2]
```

Comment:
See [maxsplit parameter](https://docs.python.org/3.3/library/stdtypes.html?highlight=set#str.split) which avoids the need for indexing later.

Reply:
I don't see the option to select only the first items when using maxsplit, or am I overseeing something? It will split until the given index but then adding all following into one item. This will cause the problem you described above:

```
>>> sqlite3_version = "4.0.0-dev"
>>> sqlite3_version.split('.',1)
['4', '0.0-dev']
>>> [int(i) for i in sqlite3_version.split('.',1)]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 1, in <listcomp>
ValueError: invalid literal for int() with base 10: '0.0-dev'
>>> sqlite3_version.split('.',2)
['4', '0', '0-dev']
>>> [int(i) for i in sqlite3_version.split('.',2)]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 1, in <listcomp>
ValueError: invalid literal for int() with base 10: '0-dev'
```

Reply:
@mmacata You're right, `maxsplit` _might_ be used to limit the resulting list to three parts, eg:

```py
>>> sqlite3_version = "4.0.0-dev"
>>> sqlite3_version.split('.', maxsplit=3)[:2]
['4', '0']
```

but the index is needed. Although that seems to me to be over doing things. Index should be enough.

Reply:
@wenzeslaus do you also agee? Can this be merged as is?